code/modules/reagents/recipes.dm

```diff
@@ -52,21 +52,30 @@
 	var/mix_message = "The solution begins to bubble."
 	var/reaction_sound = 'sound/effects/bubbles.ogg'
 
+	var/list/require_containers = list() // This reaction will only occure in these containers(Or their subtypes).
+	var/list/blacklist_containers = list() // This reaction will not occure in these containers(Or their subtypes).
+
 	var/log_is_important = 0 // If this reaction should be considered important for logging. Important recipes message admins when mixed, non-important ones just log to file.
 /datum/chemical_reaction/proc/can_happen(var/datum/reagents/holder)
 	//check that all the required reagents are present
 	if(!holder.has_all_reagents(required_reagents))
-		return 0
+		return FALSE
 
 	//check that all the required catalysts are present in the required amount
 	if(!holder.has_all_reagents(catalysts))
-		return 0
+		return FALSE
 
 	//check that none of the inhibitors are present in the required amount
 	if(holder.has_any_reagent(inhibitors))
-		return 0
+		return FALSE
+
+	if(require_containers.len && !is_type_in_list(holder.my_atom, require_containers))
```

Comment:
does this run istype on all the things in this list?

Reply:
Yes, exactly.